frontend/lib/src/components/elements/GraphVizChart/GraphVizChart.tsx

```diff
@@ -38,75 +38,55 @@ interface Dimensions {
 const dummyGraphviz = graphviz
 dummyGraphviz // eslint-disable-line @typescript-eslint/no-unused-expressions
 
+const isFullScreen = (height = 0): boolean => Boolean(height)
+
 export function GraphVizChart({
   width: propWidth,
   element,
   height: propHeight,
 }: GraphVizChartProps): ReactElement {
   const chartId = `graphviz-chart-${element.elementId}`
+  const isFull = isFullScreen(propHeight)
 
   let originalHeight = 0
   let originalWidth = 0
 
-  const getChartData = (): string => {
-    return element.spec
-  }
-
   const getChartDimensions = (): Dimensions => {
-    let chartWidth = originalWidth
-    let chartHeight = originalHeight
-
-    if (propHeight) {
-      // fullscreen
-      chartWidth = propWidth
-      chartHeight = propHeight
-    } else if (element.useContainerWidth) {
-      chartWidth = propWidth
-    }
+    const chartWidth =
+      isFull || element.useContainerWidth ? propWidth : originalWidth
+    const chartHeight = isFull ? propHeight || originalHeight : originalHeight
     return { chartWidth, chartHeight }
   }
 
-  const updateChart = (): void => {
+  const setSvgDimensions = (node: SVGGraphicsElement): void => {
+    originalHeight = Math.round(node.getBBox().height)
+    originalWidth = Math.round(node.getBBox().width)
+
+    select(node)
+      .attr("width", isFull ? `${originalWidth}pt` : "100%")
+      .attr("height", isFull ? `${originalHeight}pt` : "100%")
+  }
+
+  useEffect(() => {
     try {
       // Layout and render the graph
-      const graph = graphviz(`#${chartId}`)
-        .zoom(false)
-        .fit(true)
-        .scale(1)
-        .renderDot(getChartData())
-        .on("end", () => {
-          const node = select(`#${chartId} > svg`).node() as SVGGraphicsElement
-          if (node) {
-            originalHeight = node.getBBox().height
-            originalWidth = node.getBBox().width
-          }
-        })
-
-      const { chartHeight, chartWidth } = getChartDimensions()
-      if (chartHeight > 0) {
-        // Override or reset the graph height
-        graph.height(chartHeight)
-      }
-      if (chartWidth > 0) {
-        // Override or reset the graph width
-        graph.width(chartWidth)
-      }
+      const graph = graphviz(`#${chartId}`).zoom(false).fit(true).scale(1)
+
+      graph.renderDot(element.spec).on("end", () => {
+        const node = select(`#${chartId} > svg`).node()
+        setSvgDimensions(node as SVGGraphicsElement)
+      })
     } catch (error) {
       logError(error)
     }
-  }
 
-  useEffect(() => {
-    updateChart()
-  })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [propWidth, element.spec])
```

Comment:
should this also rerender when the height changes?

Reply:
You're right! I had excluded it because my dimensions calculation was buggy and including the height here was causing the svg height to change in unexpected cases. Fixed in subsequent commits 👍